### PR TITLE
Added Dockerfile and building instructions

### PR DESCRIPTION
**Pull request by sauliusgrigaitis**
@barnabasbusa thanks for raising this on Discord. Could you check that this is sufficient? BTW, the client still misses a few artefacts (genesis etc.) to make successful build, but the `Dockefile` and the instruction are in the readme.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM rust:1.75.0-bullseye AS builder
+RUN apt-get update && apt-get -y upgrade && apt-get install -y make libclang-dev
+COPY . grandine
+RUN cd grandine && ./scripts/build/release.sh
+
+FROM ubuntu:latest
+COPY --from=builder /usr/local/cargo/bin/grandine /usr/local/bin/grandine
```
Comment (barnabasbusa) [inline Dockerfile]: you might wanna set the default entrypoint too.

Comment (sauliusgrigaitis) [reply Dockerfile]: Just wondering is there any other benefits of default entrypoint apart from the fact that user doens't need to write "grandine" when running container?

Comment (barnabasbusa) [reply Dockerfile]: no, thats about it :D I usually assume that container --help will give me help, and wouldn't need to find the binary.

Comment (sauliusgrigaitis) [reply Dockerfile]: Laziness makes software better :)

**Comment by barnabasbusa** [comment]
```
 => ERROR [builder 4/4] RUN cd grandine && ./scripts/build/release.sh                                                                                                                                                                                                                                        23.7s 
------                                                                                                                                                                                                                                                                                                             
 > [builder 4/4] RUN cd grandine && ./scripts/build/release.sh:                                                                                                                                                                                                                                                    
0.086 info: syncing channel updates for 'stable-2024-02-08-aarch64-unknown-linux-gnu'                                                                                                                                                                                                                              
2.403 info: latest update on 2024-02-08, rust version 1.76.0 (07dca489a 2024-02-04)                                                                                                                                                                                                                                
2.403 info: downloading component 'cargo'                                                                                                                                                                                                                                                                          
3.195 info: downloading component 'clippy'                                                                                                                                                                                                                                                                         
3.409 info: downloading component 'rust-std'
6.736 info: downloading component 'rustc'
14.71 info: downloading component 'rustfmt'
15.66 info: installing component 'cargo'
16.25 info: installing component 'clippy'
16.49 info: installing component 'rust-std'
18.43 info: installing component 'rustc'
22.44 info: installing component 'rustfmt'
22.73 error: failed to load manifest for workspace member `/grandine/ad_hoc_bench`
22.73 
22.73 Caused by:
22.73   failed to load manifest for dependency `fork_choice_control`
22.73 
22.73 Caused by:
22.73   failed to load manifest for dependency `eth2_libp2p`
22.73 
22.73 Caused by:
22.73   failed to read `/grandine/eth2_libp2p/Cargo.toml`
22.73 
22.73 Caused by:
22.73   No such file or directory (os error 2)
------
Dockerfile:4
--------------------
   2 |     RUN apt-get update && apt-get -y upgrade && apt-get install -y make libclang-dev
   3 |     COPY . grandine
   4 | >>> RUN cd grandine && ./scripts/build/release.sh
   5 |     
   6 |     FROM ubuntu:latest
--------------------
ERROR: failed to solve: process "/bin/sh -c cd grandine && ./scripts/build/release.sh" did not complete successfully: exit code: 101
```

**Comment by sauliusgrigaitis** [comment]
https://github.com/grandinetech/grandine?tab=readme-ov-file#build

```
git submodule update --init
```

But even with this we still need to fix genesis files. Should be fixed very soon.

**Comment by sauliusgrigaitis** [comment]
@barnabasbusa please check when you have some time.

**Pull request closed without merging**